### ISSUE TITLE
The NF should simplify conditional expressions in non-scalarized bindings

### ISSUE THREAD

**Issue by casella** [OpenModelica/OpenModelica#9935]
I flattened model ScalableTestGrids.Models.Type1_N_1_M_1 with flags
```
-f -d=nonfScalarize,mergeComponents,evaluateAllParameters,vectorizeBindings --frontendInline
```
and I got lines such as 
```modelica
Real[2] '$TransformerFixedRatio2.portB.UPhase'(
  start = {0.0, 0.0},
  displayUnit = fill("deg", 2),
  unit = fill("rad", 2), 
  quantity = fill("Angle", 2)) = 
    {if {true, true}['$$TransformerFixedRatio21'] then
      atan2('$TransformerFixedRatio2['$$TransformerFixedRatio21'].portB.v'.'im',     
            '$TransformerFixedRatio2['$$TransformerFixedRatio21'].portB.v'.'re')
     else 0.0 
     for $$TransformerFixedRatio21 in 1:2} "Phase of voltage across the port";
```
This looks quite complicated to handle by an array-preserving backend. However, `{true, true}['$$TransformerFixedRatio21']` always has value `true` whatever the index `$$TransformerFixedRatio21`, so the if expression could be easily simplified as
```modelica
Real[2] '$TransformerFixedRatio2.portB.UPhase'(
  start = {0.0, 0.0},
  displayUnit = fill("deg", 2),
  unit = fill("rad", 2), 
  quantity = fill("Angle", 2)) = 
    {atan2('$TransformerFixedRatio2['$$TransformerFixedRatio21'].portB.v'.'im', 
            '$TransformerFixedRatio2['$$TransformerFixedRatio21'].portB.v'.'re')
     for $$TransformerFixedRatio21 in 1:2} "Phase of voltage across the port";
```
Something similar could be done if we had condition if {false, false}['$$TransformerFixedRatio21'], which would lead to 
```modelica
Real[2] '$TransformerFixedRatio2.portB.UPhase'(
  start = {0.0, 0.0},
  displayUnit = fill("deg", 2),
  unit = fill("rad", 2), 
  quantity = fill("Angle", 2)) = 
    {0.0 for $$TransformerFixedRatio21 in 1:2} "Phase of voltage across the port";
```
or, even better,
```modelica
Real[2] '$TransformerFixedRatio2.portB.UPhase'(
  start = {0.0, 0.0},
  displayUnit = fill("deg", 2),
  unit = fill("rad", 2), 
  quantity = fill("Angle", 2)) = fill(0.0, 2) "Phase of voltage across the port";
```
@perost could you do that in the NF when the condition array being indexed has all equal elements, so you can decide what it the active if branch on a wholesale array basis?

**Comment by perost** [comment]
Fixed in #9941. I added a general simplification for subscripted expressions rather than something only for conditionals, since it's easier to be general in this case. There's also a lot of such expressions in the flat model, so hopefully that will help.

Our CI seems to be having some issues right now though, so it might take a while before the PR is merged.

**Comment by casella** [comment]
> Fixed in #9941. I added a general simplification for subscripted expressions rather than something only for conditionals, since it's easier to be general in this case. There's also a lot of such expressions in the flat model, so hopefully that will help.

Very good, thanks!

> Our CI seems to be having some issues right now though, so it might take a while before the PR is merged.

The Windows nightlies worked fine until last night. Linux builds are very fragile, but that's not new. What issues are you referring to?

**Comment by perost** [comment]
> > Our CI seems to be having some issues right now though, so it might take a while before the PR is merged.
> 
> The Windows nightlies worked fine until last night. Linux builds are very fragile, but that's not new. What issues are you referring to?

Jenkins would fail already at the first step of checking out the PR, so my PR isn't merged yet. I restarted it again, looks like it might be working better now.
